File: src/cli/terminal.ts
/** Shared terminal dimension helpers used by render and watch modules. */

export type Density = "wide" | "medium" | "narrow";

export function resolveWidth(raw?: number): number {
  if (typeof raw === "number" && Number.isFinite(raw) && raw > 0) {
    return Math.floor(raw);
  }
  if (typeof process.stdout.columns === "number" && process.stdout.columns > 0) {
    return process.stdout.columns;
  }
  return 120;
}

export function resolveDensity(width: number): Density {
  if (width >= 120) return "wide";
  if (width >= 90) return "medium";
  return "narrow";
}

File: src-rust/src/cli/terminal.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum Density {
    Wide,
    Medium,
    Narrow,
}

pub fn resolve_width(raw: Option<usize>) -> usize {
    if let Some(width) = raw {
        if width > 0 {
            return width;
        }
    }
    if let Some((terminal_size::Width(width), _)) = terminal_size::terminal_size() {
        if width > 0 {
            return width as usize;
        }
    }
    120
}

pub fn resolve_density(width: usize) -> Density {
    if width >= 120 {
        Density::Wide
    } else if width >= 90 {
        Density::Medium
    } else {
        Density::Narrow
    }
}
